### DISCUSSION TITLE
订阅功能如何使用？

### DISCUSSION THREAD

**Discussion by asd2003asd** [kkkgo/PaoPaoGateWay#77] [问和答]
ppgw.ini文件已经放入ppdns deta文件夹，日志显示老是从http://192.168.0.10:7889/ppgw.ini  ros主路由获取配置文件。

**Comment by kkkgo** [comment]
https://github.com/kkkgo/PaoPaoGateWay/blob/f80f13c1d6466e9253b822eeac4ce20a25af01a1/ReadMe.md?plain=1#L26-L37

**Comment by asd2003asd** [comment]
![photo_2024-03-22_14-11-26](https://github.com/kkkgo/PaoPaoGateWay/assets/19873779/7355f189-14db-4199-81b6-8de638cb671a)

**Reply by asd2003asd** [reply]
version: "3.5"

services:
  paopaodns:
    image: sliamb/paopaodns:latest
    container_name: PaoPaoDNS
    restart: always
    volumes:
      - /mnt/docker/paopaodns:/data
    environment:
      - TZ=Asia/Shanghai
      - UPDATE=weekly
      - DNS_SERVERNAME=PaoPaoDNS,blog.03k.org
      - DNSPORT=53
      - CNAUTO=yes
      - CNFALL=yes
      - CN_TRACKER=yes
      - USE_HOSTS=no
      - IPV6=no
      - SOCKS5=192.168.0.30:1080
      - SERVER_IP=192.168.0.99
      - CUSTOM_FORWARD=192.168.0.30:53
      - AUTO_FORWARD=yes
      - AUTO_FORWARD_CHECK=yes
      - HTTP_FILE=yes
    ports:
      - "53:53/udp"
      - "53:53/tcp"
      - "5304:5304/udp"
      - "5304:5304/tcp"
      - "7889:7889/tcp"

**Comment by asd2003asd** [comment]
[ppgw.zip](https://github.com/kkkgo/PaoPaoGateWay/files/14718321/ppgw.zip)

**Comment by asd2003asd** [comment]
大佬再帮忙指导一下，我现在用内置custom.yaml文件的方式可以正常使用，现在想用机场的订阅链接定时更新节点信息。

**Comment by kkkgo** [comment]
1、你给虚拟机分配的DNS是PaoPaoDNS吗？不是肯定拉不了配置。
2、你内置custom.yaml能用那你内置ppgw.ini的suburl也能用。
3、你自己都可以打开浏览器看看你的配置到底能不能拉下来。

**Reply by kkkgo** [reply]
文档写的很清楚了，不内置配置的就是从你网卡的DNS拉的，你自己都可以尝试用浏览器看看能不能打开7889/ppgw.ini，你又不贴你的网卡的信息贴这些配置有啥用

**Reply by asd2003asd** [reply]
ip=192.168.0.30
mask=255.255.255.0
gw=192.168.0.10
dns=192.168.0.99

**Reply by kkkgo** [reply]
你这是docker定制的配置还是你dhcp下发的配置，你是docker定制的话很明显写错了

**Reply by asd2003asd** [reply]
docker定制的

**Reply by kkkgo** [reply]
https://github.com/kkkgo/PaoPaoGateWay/blob/f80f13c1d6466e9253b822eeac4ce20a25af01a1/ReadMe.md?plain=1#L154-L161

**Reply by asd2003asd** [reply]
dns2 该配置哪个？我就安了一个ppdns

**Reply by kkkgo** [reply]
没有就不填

**Comment by asd2003asd** [comment]
![1](https://github.com/kkkgo/PaoPaoGateWay/assets/19873779/1094bc84-56d4-4c38-8532-88882aa21374)
现在配置文件能读取了，节点配置文件无法生成，订阅链接复制到浏览器可以获取到节点。

**Reply by kkkgo** [reply]
你用yaml模式你不放yaml文件哪来的配置文件？

**Reply by asd2003asd** [reply]
放个空的custom.yaml文件可以吗？我想用订阅链接的节点。

**Reply by kkkgo** [reply]
你想用订阅就不要用yaml模式用suburl模式啊，看文档。
https://github.com/kkkgo/PaoPaoGateWay/blob/f80f13c1d6466e9253b822eeac4ce20a25af01a1/ReadMe.md?plain=1#L114-L119

**Reply by asd2003asd** [reply]
好的，谢谢啦。用suburl模式能正常订阅使用了。
